flink-runtime/src/main/java/org/apache/flink/runtime/io/network/NetworkEnvironment.java

```diff
@@ -202,28 +233,97 @@ public TaskKvStateRegistry createKvStateTaskRegistry(JobID jobId, JobVertexID jo
 		return kvStateRegistry.createTaskRegistry(jobId, jobVertexId);
 	}
 
+	public ResultPartition[] getResultPartitionWriters(String taskId) {
+		return allPartitions.get(taskId);
+	}
+
+	public SingleInputGate[] getInputGates(String taskId) {
```

Comment:
The returned type could be changed to `InputGate[]` and `ResultPartitionWriter[]` after #7767 , #7785  merged.